install.zasilkovna.php

```diff
@@ -130,63 +178,48 @@ public function postflight($route, JAdapterInstance $adapter) {
             }
 
 			$db = JFactory::getDBO();
-			$q = "CREATE TABLE IF NOT EXISTS #__virtuemart_zasilkovna_branches (
-										`id` int(10) NOT NULL,
-										`name_street` varchar(200) NOT NULL,
-										`currency` text NOT NULL,
-										`country` varchar(10) NOT NULL
-										) ENGINE=MyISAM DEFAULT CHARSET=utf8;";
-			$db->setQuery($q);
-			$db->query();
-
-			$q = "CREATE TABLE IF NOT EXISTS `#__virtuemart_shipment_plg_zasilkovna` (
-							  `id` int(1) unsigned NOT NULL AUTO_INCREMENT,
-							  `virtuemart_order_id` int(11) unsigned DEFAULT NULL,
-							  `virtuemart_shipmentmethod_id` mediumint(1) unsigned DEFAULT NULL,
-							  `order_number` char(32) DEFAULT NULL,
-							  `zasilkovna_packet_id` decimal(10,0) DEFAULT NULL,
-							  `zasilkovna_packet_price` decimal(15,2) DEFAULT NULL,
-							  `weight` decimal(10,4) DEFAULT NULL,
-							  `branch_id` decimal(10,0) DEFAULT NULL,
-							  `branch_currency` char(5) DEFAULT NULL,
-							  `branch_name_street` varchar(500) DEFAULT NULL,
-							  `is_carrier` smallint(1) NOT NULL DEFAULT '0',
-							  `carrier_pickup_point` varchar(40) DEFAULT NULL,
-							  `email` varchar(255) DEFAULT NULL,
-							  `phone` varchar(255) DEFAULT NULL,
-							  `first_name` varchar(255) DEFAULT NULL,
-							  `last_name` varchar(255) DEFAULT NULL,
-							  `address` varchar(255) DEFAULT NULL,
-							  `city` varchar(255) DEFAULT NULL,
-							  `zip_code` varchar(255) DEFAULT NULL,
-							  `virtuemart_country_id` varchar(255) DEFAULT NULL,
-							  `adult_content` smallint(1) DEFAULT '0',
-							  `is_cod` smallint(1) DEFAULT NULL,
-							  `packet_cod` decimal(15,2) DEFAULT '0',
-							  `exported` smallint(1) DEFAULT NULL,
-							  `printed_label` smallint(1) DEFAULT '0',
-							  `shipment_name` varchar(5000) DEFAULT NULL,
-							  `shipment_cost` decimal(10,2) DEFAULT NULL,
-							  `shipment_package_fee` decimal(10,2) DEFAULT NULL,
-							  `tax_id` smallint(1) DEFAULT NULL,
-							  `created_on` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-							  `created_by` int(11) NOT NULL DEFAULT '0',
-							  `modified_on` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-							  `modified_by` int(11) NOT NULL DEFAULT '0',
-							  `locked_on` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-							  `locked_by` int(11) NOT NULL DEFAULT '0',
-							  PRIMARY KEY (`id`)
-							) ENGINE=MyISAM DEFAULT CHARSET=utf8 COMMENT='zasilkovna';";
-			$db->setQuery($q);
-			$db->query();
-
 			$q = "
 INSERT INTO #__virtuemart_adminmenuentries (`module_id`, `parent_id`, `name`, `link`, `depends`, `icon_class`, `ordering`, `published`, `tooltip`, `view`, `task`) VALUES
 							(5, 0, 'ZASILKOVNA', '', '', 'vmicon vmicon-16-zasilkovna', 1, 1, '', 'zasilkovna', '');";
 			$db->setQuery($q);
-			$db->query();
+			$db->execute();
 
 		}
+
+        if ($route === 'update' && $this->fromVersion && version_compare($this->fromVersion, '1.2.0') < 0) {
```

Comment:
Nebylo by použití operátoru '<' přehlednější?

Reply:
Neřekl bych. V čem je to přehlednější?

Reply:
version_compare potom vrací boolean, což  je to, co chceme.

Reply:
My ale nechceme, aby funkce vracela boolean. My chceme funkci, která nějakým způsobem porovná softwarové verze a vrátí hodnotu, která zachycuje rozdílnost verzí.

Boolean bych čekal od funkce, která je pojmenovaná ve stylu otázky. Jako třeba file_exists() nebo is_array().

Reply:
upraveno